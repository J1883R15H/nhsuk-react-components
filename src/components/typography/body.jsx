import React from 'react';
import { oneOf, node, string } from 'prop-types';
import stylePropType from 'react-style-proptype';

const Body = ({ size, children, className, style }) => {
  return (
    <p
      className={`nhsuk-body${
        size && size !== 'm' ? `-${size}` : ''
      } ${className}`}
      style={style}
    >
      {children}
    </p>
  );
};
Body.propTypes = {
  size: oneOf(['s', 'm', 'l']),
  children: node.isRequired,
  style: stylePropType,
  className: string
};

Body.defaultProps = {
  size: 'm',
  style: {},
  className: ''
};

export default Body;
