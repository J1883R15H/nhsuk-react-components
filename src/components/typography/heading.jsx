import React from 'react';
import { oneOf, node, string } from 'prop-types';
import classNames from 'classnames';

const Heading = ({ size, children, className, ...rest }) => {
  switch (size) {
    case 'xl':
      return (
        <h1 className={classNames('nhsuk-heading-xl', className)} {...rest}>
          {children}
        </h1>
      );
    case 'l':
      return (
        <h2 className={classNames('nhsuk-heading-l', className)} {...rest}>
          {children}
        </h2>
      );
    case 'm':
      return (
        <h3 className={classNames('nhsuk-heading-m', className)} {...rest}>
          {children}
        </h3>
      );
    case 's':
      return (
        <h4 className={classNames('nhsuk-heading-s', className)} {...rest}>
          {children}
        </h4>
      );
    case 'xs':
      return (
        <h5 className={classNames('nhsuk-heading-xs', className)} {...rest}>
          {children}
        </h5>
      );
    default:
      return null;
  }
};

Heading.propTypes = {
  size: oneOf(['xl', 'l', 'm', 's', 'xs']),
  children: node.isRequired,
  className: string
};

Heading.defaultProps = {
  size: 'xl',
  className: ''
};

export default Heading;
