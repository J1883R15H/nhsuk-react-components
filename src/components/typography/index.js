import Body from './body';
import Heading from './heading';
import List from './list';

export { Body, Heading, List };
