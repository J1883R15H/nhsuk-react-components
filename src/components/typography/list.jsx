import React from 'react';
import { oneOf, node, string } from 'prop-types';
import stylePropType from 'react-style-proptype';

const List = ({ type, children, className, style }) => {
  if (type === 'number') {
    return (
      <ol
        className={`nhsuk-list nhsuk-list--number ${className}`}
        style={style}
      >
        {children}
      </ol>
    );
  }
  return (
    <ul className={`nhsuk-list nhsuk-list--bullet ${className}`} style={style}>
      {children}
    </ul>
  );
};

List.propTypes = {
  type: oneOf(['bullet', 'number']),
  children: node.isRequired,
  className: string,
  style: stylePropType
};

List.defaultProps = {
  type: 'bullet',
  className: '',
  style: {}
};
