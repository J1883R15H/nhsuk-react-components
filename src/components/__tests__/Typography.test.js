import React from 'react';
import { shallow, mount } from 'enzyme';
import { Body, Heading, List } from '../typography';

describe('Typography', () => {
  describe('Body', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(<Body>Text</Body>);
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
    it('has proper styling', () => {
      const smallBody = mount(<Body size="s">Text</Body>);
      const mediumBody = mount(<Body size="m">Text</Body>);
      const largeBody = mount(<Body size="l">Text</Body>);
      expect(
        smallBody.containsMatchingElement(<p className="nhsuk-body-s">Text</p>)
      ).toBeTruthy();
      expect(
        mediumBody.containsMatchingElement(<p className="nhsuk-body">Text</p>)
      ).toBeTruthy();
      expect(
        largeBody.containsMatchingElement(<p className="nhsuk-body-l">Text</p>)
      ).toBeTruthy();
      largeBody.unmount();
      mediumBody.unmount();
      smallBody.unmount();
    });
  });
  describe('Heading', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(<Heading>Text</Heading>);
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
    it('renders proper heading on size supplied', () => {
      const xl = shallow(<Heading size="xl">Title</Heading>);
      const l = shallow(<Heading size="l">Title</Heading>);
      const m = shallow(<Heading size="m">Title</Heading>);
      const s = shallow(<Heading size="s">Title</Heading>);
      const xs = shallow(<Heading size="xs">Title</Heading>);
      expect(
        xl.containsMatchingElement(<h1 className="nhsuk-heading-xl">Title</h1>)
      ).toBeTruthy();
      expect(
        l.containsMatchingElement(<h2 className="nhsuk-heading-l">Title</h2>)
      ).toBeTruthy();
      expect(
        m.containsMatchingElement(<h3 className="nhsuk-heading-m">Title</h3>)
      ).toBeTruthy();
      expect(
        s.containsMatchingElement(<h4 className="nhsuk-heading-s">Title</h4>)
      ).toBeTruthy();
      expect(
        xs.containsMatchingElement(<h5 className="nhsuk-heading-xs">Title</h5>)
      ).toBeTruthy();
      xl.unmount();
      l.unmount();
      m.unmount();
      s.unmount();
      xs.unmount();
    });
    it('returns null (for coverage)', () => {
      const invalidSize = shallow(<Heading size="xxl">Title</Heading>);
      expect(
        invalidSize.containsMatchingElement(
          <h1 className="nhsuk-heading-xl">Title</h1>
        )
      ).toBeFalsy();
      expect(invalidSize.getElement()).toEqual(null);
      invalidSize.unmount();
    });
  });
  describe('List', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(
        <List>
          <li>Text</li>
        </List>
      );
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
    it('returns an ordered list', () => {
      const shallowMountNumber = shallow(
        <List type="number">
          <li>Text</li>
        </List>
      );
      expect(
        shallowMountNumber.containsMatchingElement(
          <ol className="nhsuk-list nhsuk-list--number">
            <li>Text</li>
          </ol>
        )
      ).toBeTruthy();
      shallowMountNumber.unmount();
      const shallowMountUnordered = shallow(
        <List type="bullet">
          <li>Text</li>
        </List>
      );
      expect(
        shallowMountUnordered.containsMatchingElement(
          <ul className="nhsuk-list nhsuk-list--bullet">
            <li>Text</li>
          </ul>
        )
      ).toBeTruthy();
      shallowMountUnordered.unmount();
    });
  });
});
