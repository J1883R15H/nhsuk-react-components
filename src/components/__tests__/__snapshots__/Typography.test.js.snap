// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Body matches snapshot 1`] = `
<p
  className="nhsuk-body "
  style={Object {}}
>
  Text
</p>
`;

exports[`Typography Heading matches snapshot 1`] = `
<h1
  className="nhsuk-heading-xl "
  style={Object {}}
>
  Text
</h1>
`;

exports[`Typography List matches snapshot 1`] = `
<ul
  className="nhsuk-list nhsuk-list--bullet "
  style={Object {}}
>
  <li>
    Text
  </li>
</ul>
`;
